tests/react/async.test.tsx

```diff
@@ -52,43 +53,44 @@ it('does not show async stale result', async () => {
     return <div>delayedCount: {delayedCount}</div>
   }
 
-  await act(async () => {
+  await act(() =>
     render(
       <>
         <Counter />
-        <Suspense fallback="loading">
+        <Suspense fallback={<div>loading</div>}>
           <DelayedCounter />
         </Suspense>
       </>,
-    )
-  })
+    ),
+  )
+
+  expect(screen.getByText('loading')).toBeInTheDocument()
+  await act(() => vi.advanceTimersByTimeAsync(100))
+  expect(screen.getByText('count: 0')).toBeInTheDocument()
+  expect(screen.getByText('delayedCount: 0')).toBeInTheDocument()
 
-  expect(await screen.findByText('loading')).toBeInTheDocument()
-  resolve1()
-  resolve2()
-  expect(await screen.findByText('count: 0')).toBeInTheDocument()
-  expect(await screen.findByText('delayedCount: 0')).toBeInTheDocument()
   expect(committed).toEqual([0])
 
-  await userEvent.click(screen.getByText('button'))
-  expect(await screen.findByText('loading')).toBeInTheDocument()
-  await act(async () => {
-    resolve1()
-    resolve2()
-    await Promise.resolve()
-    resolve2()
-  })
-  expect(await screen.findByText('count: 2')).toBeInTheDocument()
-  expect(await screen.findByText('delayedCount: 2')).toBeInTheDocument()
-  expect(committed).toEqual([0, 2])
+  await act(() => fireEvent.click(screen.getByText('button')))
+  expect(screen.getByText('loading')).toBeInTheDocument()
+  await act(() => vi.advanceTimersByTimeAsync(100))
+  await act(() => vi.advanceTimersByTimeAsync(100))
+  expect(screen.getByText('count: 2')).toBeInTheDocument()
+  expect(screen.getByText('delayedCount: 2')).toBeInTheDocument()
+
+  // React 18+ uses automatic batching, so committed is [0, 2]
+  // React 16-17 doesn't batch async updates, so committed is [0, 1, 2]
+  // Different build types (cjs, umd, esm) may also affect batching behavior
```

Comment:
Thanks for the comment. This should be good. We'll dropping old React versions in the major version.